### PR TITLE
FEATURE: Added a FileStorage that serializes instances to a JSON file and deserializes JSON file to instances

### PR DESCRIPTION
**Pull request by thenoblet**
### Overview
This pull request introduces a robust and versatile `FileStorage` class, designed to manage the serialization and deserialization of instances to and from a JSON file. The implementation is located in `models/engine/file_storage.py`. Furthermore, it seamlessly integrates into the application through necessary updates in `models/__init__.py` and `models/base_model.py`.

#### FileStorage Class (models/engine/file_storage.py)
- **Private Class Attributes:**
  - `__file_path`: A string representing the path to the JSON file. For instance, `file.json`.
  - `__objects`: A dictionary intended to store all objects by `<class name>.id` (e.g., `BaseModel.12121212`).

- **Public Instance Methods:**
  - `all(self)`: Returns the dictionary `__objects`.
  - `new(self, obj)`: Sets `obj` in `__objects` with the key `<obj class name>.id`.
  - `save(self)`: Serializes `__objects` to the JSON file specified by `__file_path`.
  - `reload(self)`: Deserializes the JSON file to `__objects` only if the file exists. If the file doesn't exist, no exception is raised.

#### models/__init__.py
- Updates to create a unique `FileStorage` instance for the application:
  ```python
  from models.engine.file_storage import FileStorage
  storage = FileStorage()
  storage.reload()

### CODE REVIEW

File: models/engine/file_storage.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                instances = json.load(file)
+
+            for key, obj_dict in instances.items():
+                # Extract class name and object id from the key
+                class_name, obj_id = key.split(".")
+
+                # Dynamically import the class and create an instance
+                """ obj_instance = getattr(
+                __import__(class_name), class_name)(**obj_dict) """
+                obj_instance = self.__models[class_name](**obj_dict)
```
Comment (LawsonLawson) [inline models/engine/file_storage.py:93]: Are we keeping the dynamic-import code? else I suggest we remove it.

**Review by LawsonLawson — COMMENTED** [review]
Aside the dynamic import code you commented out, I think everything else looks good boss.

**Comment by thenoblet** [comment]
We good!

**Comment by thenoblet** [comment]
> Aside the dynamic import code you commented out, I think everything else looks good boss.

I will test come things with the commented out code. I shall remove it later.

**Pull request merged**